tensorzero-core/src/config_parser/mod.rs

```diff
@@ -1275,21 +1295,31 @@ pub enum UninitializedVariantConfig {
     ChainOfThought(UninitializedChainOfThoughtConfig),
 }
 
+/// Holds extra information used for enriching error messages
+pub struct ErrorContext {
```

Comment:
this is a smell because of our log-on-construction behavior. It's OK for now but I've opened an issue to track us fixing this. #3169